src/main/java/spoon/support/compiler/jdt/JDTBatchCompiler.java

```diff
@@ -63,9 +66,36 @@
 		}
 	}
 
-	/** we force this method of JDT to be re-implemented, see subclasses */
 	@Override
-	public abstract CompilationUnit[] getCompilationUnits();
+	public CompilationUnit[] getCompilationUnits() {
+		return compilationUnits;
+	}
+
+	public void setCompilationUnits(CompilationUnit[] compilationUnits) {
+		this.compilationUnits = compilationUnits;
+	}
+
+	public void setInputFiles(List<SpoonFile> files) {
```

Comment:
I propose that this method moves to FileCompilerConfig. And FileCompilerConfig would be responsible for calling setCompilationUnits().